Actions/WorkflowPostProcess/WorkflowPostProcess.ps1

```diff
@@ -1,26 +1,61 @@
 ﻿Param(
-    [Parameter(HelpMessage = "The event Id of the initiating workflow", Mandatory = $true)]
-    [string] $eventId,
     [Parameter(HelpMessage = "Telemetry scope generated during the workflow initialization", Mandatory = $false)]
-    [string] $telemetryScopeJson = '7b7d'
+    [string] $telemetryScopeJson = '',
+    [Parameter(HelpMessage = "The current job context", Mandatory = $false)]
+    [string] $currentJobContext = ''
 )
 
-$telemetryScope = $null
-
-try {
-    . (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
-    DownloadAndImportBcContainerHelper
-    import-module (Join-Path -path $PSScriptRoot -ChildPath "..\TelemetryHelper.psm1" -Resolve)
+function GetWorkflowConclusion($JobContext) {
+    # Check the conclusion for the current job
+    if ($JobContext -ne '') {
+        $jobContext = $JobContext | ConvertFrom-Json
+        if ($jobContext.status -eq 'failure') {
+            return "Failure"
+        }
+    }
 
-    Write-Host "Post-processing workflow $eventId"
-    if ($telemetryScopeJson -and $telemetryScopeJson -ne '7b7d') {
-        $telemetryScope = RegisterTelemetryScope (hexStrToStr -hexStr $telemetryScopeJson)
-        TrackTrace -telemetryScope $telemetryScope
+    # Check the conclusion for the past jobs in the workflow
+    $workflowJobs = gh api /repos/$ENV:GITHUB_REPOSITORY/actions/runs/$ENV:GITHUB_RUN_ID/jobs -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" | ConvertFrom-Json
```

Comment:
This would return only 30 jobs: https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#list-jobs-for-a-workflow-run

Reply:
Added --paginate to the request